projects/igniteui-angular/src/lib/grids/cell.component.ts

```diff
@@ -919,4 +919,14 @@ export class IgxGridCellComponent implements OnInit, OnChanges, OnDestroy {
         return Math.max(...Array.from(this.nativeElement.children)
             .map((child) => getNodeSizeViaRange(range, child)));
     }
+
+    /**
+     * @hidden
+     * @internal
+     */
+    public get searchMetadata() {
+        const meta = new Map<string, any>();
+        meta.set('dataIndex', (this.row as any).viewIndex);
```

Comment:
Rather than cast to any it would be better to have the viewIndex defined in the RowType as it is required for search to work correctly with pinned rows in all grids.